tt_metal/impl/sub_device/sub_device_manager.cpp

```diff
@@ -156,6 +156,16 @@ bool SubDeviceManager::has_allocations() const {
 
 DeviceAddr SubDeviceManager::local_l1_size() const { return this->local_l1_size_; }
 
+void SubDeviceManager::set_fabric_sub_device_id(SubDeviceId fabric_sub_device_id) {
+    const auto& fabric_sub_device = this->sub_device(fabric_sub_device_id);
+    TT_FATAL(
+        fabric_sub_device.cores(HalProgrammableCoreType::TENSIX).num_cores() == 0,
+        "Fabric sub device must not have Tensix cores");
+    this->fabric_sub_device_id_ = fabric_sub_device_id;
+}
+
+std::optional<SubDeviceId> SubDeviceManager::fabric_sub_device_id() const { return this->fabric_sub_device_id_; }
```

Comment:
I think we can start dropping `this->` if the member is named with the `_` suffix, and there is no ambiguity - wdyt?

Reply:
Sounds good. We should add a style doc for people to reference.